### ISSUE TITLE
When recording with "Use GPU Recording" on, it is required to toggle "Threaded Video" to get smooth gameplay

### ISSUE THREAD

**Issue by thingsiplay** [libretro/RetroArch#14013]
# First and foremost consider this:

I have this problem and found a workaround. But it is not a solution in my opinion and needs to be addressed in RetroArch itself. The problem happens regardless of the core. Included are files with my system information, RetroArch output and Nvenc configuration.

```
RetroArch: 1.10.3 (Git 9b282aa742) - Built: Apr 26 2022
OS: Manjaro Linux [x86_64]
Kernel: 5.17.9-1-MANJARO
[Vulkan]: Found GPU at index 0: "NVIDIA GeForce GTX 1070".
Vulkan]: Found vulkan context: "vk_x".
[Vulkan]: Detecting screen resolution: 2560x1440.
[X/Vulkan]: Window manager is LG3D.
[Video]: Average monitor Hz: 143.760782 Hz. (4.017 % frame time deviation, based on 2048 last samples).
```

## Description

When starting the video recording to file at full resolution (1440p) with a shader and "Use GPU Recording" is enabled (Nvenc), then it is very choppy to play the game from that point. As if my PC could not handle it, but it should and it can. The produced video file is playing just fine, it is just while recording gaming is almost impossible with slow and choppy play.

But the strange thing is, when I (while recording) go to RetroArch menu and toggle option "Threaded Video" to ON or OFF and back or not to previous setting, as long as it is toggled once, then the gaming session becomes flawless. The above problem goes away and the recording is still happening. The resulting file is still good.

### Expected behavior

Start recording without the need to toggle option "Threaded Video" to get smooth gameplay while recording is happening.

### Actual behavior

Stutter and choppy gameplay, unless I toggle "Threaded Video".

### Steps to reproduce the bug

Option "Use GPU Recording" should be enabled in menu "Recording" prior the below steps.

1. Start any core.
2. Start recording video file.
3. Toggle option "Threaded Video" to get smooth gameplay while recording.

### Bisect Results

It seems to be an old bug, but I only encountered lately because I never tried to record with RetroArch prior to this. I became aware of the issue a 2 months ago or something like that here: https://github.com/libretro/RetroArch/issues/7841

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.10.3/9b282aa742

### Environment information

- OS: Manjaro
- Compiler: ---

[mysyestem.txt](https://github.com/libretro/RetroArch/files/8842746/mysyestem.txt)
[retroarch-output_recording.txt](https://github.com/libretro/RetroArch/files/8842754/retroarch-output_recording.txt)
[nvidia-mysettings.cfg.txt](https://github.com/libretro/RetroArch/files/8842755/nvidia-mysettings.cfg.txt)

**Comment by MestreLion** [comment]
Sir @thingsiplay , you're a lifesaver!!!

I've stumbled on this behavior by chance, while adjusting a dozen settings to fix GPU recording. Thanks to you I was able to isolate and reproduce the exact steps needed to fix the choppy gameplay. And it behaves _exactly_ like you described!

Also, toggling Threaded GPU on and off _during_ a recording also fixes the gameplay for all subsequent recordings, so I only have to toggle it once per RetroArch session, and not per-recording.

Using Batocera 37, Retroarch 1.15.0 on an Intel N100 Mini PC (which I'm *so* glad its modest iGPU is able to handle xBRZ 4X with SNES9x _and_ GPU recording!)

**Comment by thingsiplay** [comment]
Hi. Glad it helped and thanks for confirming. Now that means its not only my system or hardware, as I thought it might be Nvidia related.

**Comment by ChrisWNY** [comment]
Has this issue been addressed/fixed? I've noticed the problem still lingers in the most current (stable) version of Retroarch. I have to toggle threaded video ON in order for the game play to smooth out while GPU recording is enabled.

Also, when using Parallel64 as a core, the ONLY way I can record is when GPU recording is on, and the video comes out at a super high framerate and is out of sync with the audio. I'm using "HIGH" as the video quality, files saving to MKV (default).

**Comment by thingsiplay** [comment]
No, it's not fixed for me either; and I have completely new hardware. I wonder if a workaround could be implemented (maybe a toggle to enable/disable the workaround), where when the recording is started then it automatically quickly toggles Threaded video twice. Or at least give us a hotkey to toggle Threaded video, then I can create a system wide hotkey to combine recording and threaded in one keypress. Just an idea.

**Comment by retropieuser** [comment]
Definitely not fixed. Testing on a raspberry Pi 5 the streaming functionality. Cores that need GPU Recording are a real pain

**Comment by RetroBreak** [comment]
I'm also having the same issue when trying to record using the ROG Ally, toggling threaded video fixes it, but it's a pain to have to do that every time..

**Comment by f8less** [comment]
Same issue, still happening with RA 1.20.0 and the latest FBNeo core.  FPS numbers jump into the 10s of millions once recording has started, until Threaded Video has been toggled ON/OFF -or- OFF/ON (depending on your situation), and then returns to normal levels after.

A fix for this would be greatly appreciated, or at least having NON-Use GPU Recording adhering to OS screen orientation when recording.  I play on a portrait screen orientation 100% of the time, yet RA seems to ignore the OS rotation and outputs NON-Use GPU Recording videos, rotated 90 degrees.  "Use GPU Recording" is my only current solution for this, but the OP hurdle still exists, creating extra steps in order to record game runs.
